ocaml/xenopsd/xc/service.ml

```diff
@@ -294,26 +310,27 @@ module DaemonMgmt (D : DAEMONPIDPATH) = struct
                   None
                 with Unix.Unix_error (Unix.EAGAIN, _, _) ->
                   (* cannot obtain lock: process is alive *)
-                  debug "%s: pid get from pidfile: %d" D.name pid ;
+                  pid_source := "pidfile" ;
                   Some pid
             )
         | Xenstore key | Path_of {key; _} ->
             (* case 1: backward compatibility during update installation: only has
                  xenstore pid available.
                case 2: pidfile(file domid) got removed unexpectly *)
             let pid = xs.Xs.read (key domid) in
-            debug "%s: pid get from xs: %s" D.name pid ;
+            pid_source := "xenstore" ;
             Some (int_of_string pid)
         | _ ->
             None
       with _ -> None
     in
-    if is_cmdline_valid ~pid (D.expected_cmdline_items ~domid) then
+    if
+      is_cmdline_valid ~pid ~pid_source:!pid_source
```

Comment:
```suggestion
      is_cmdline_valid ~pid ~pid_source
```

Reply:
I don't think so, because of the `!`.

Reply:
Oh I see this is part of removing the `ref`.